Sources/PackageCollections/Model/Collection.swift

```diff
@@ -71,52 +71,69 @@ extension PackageCollectionsModel {
 
 extension PackageCollectionsModel {
     /// Represents the source of a `PackageCollection`
-    public enum PackageCollectionSource: Equatable {
-        /// Package feed at URL
-        case feed(URL)
+    public struct CollectionSource: Equatable, Hashable, Codable {
+        /// Source type
+        public let type: CollectionSourceType
+
+        /// URL of the source file
+        public let url: URL
+
+        public init(type: CollectionSourceType, url: URL) {
+            self.type = type
+            self.url = url
+        }
     }
-}
 
-extension PackageCollectionsModel.PackageCollectionSource: Codable {
-    public enum DiscriminatorKeys: String, Codable {
+    /// Represents the source type of a `PackageCollection`
+    public enum CollectionSourceType: Equatable, CaseIterable {
         case feed
     }
+}
 
+extension PackageCollectionsModel.CollectionSourceType: Codable {
```

Comment:
Is the whole purpose of this just to be able to throw `UnknownType` error?

Reply:
@yim-lee are you asking about the purpose of `CollectionSourceType`? If so, it for us to be able to support multiple type of providers. e.g. feed, index, etc

Reply:
Can we use synthesized conformance and remove this extension?

Reply:
ah, I dont believe enums synthesize conformance. am I missing anything?

Reply:
It could if `CollectionSourceType: String, Codable` but not sure we want to add `: String` though.